src/Events/Dispatcher.php

```diff
@@ -30,8 +32,12 @@ class Dispatcher extends BaseDispatcher
      * @param  int  $priority
      * @return void
      */
-    public function listen($events, $listener, $priority = 0)
+    public function listen($events, $listener = null, $priority = 0)
     {
+        if ($events instanceof Closure || $events instanceof QueuedClosure) {
+            return parent::listen($events, $listener, $priority);
```

Comment:
I think we need to actually handle the processing of the closure events ourselves as Laravel does not include support for event priorities in their event dispatcher. Thoughts?

Reply:
That's a lot of rewriting I would gather, as everything seems to be string orientied and would neeed to be refactored to account for the closures and the QueuedClosures. 
The latter are nightmarish for testing, as they require the event bus to be up and running, and I have tried and failed getting that working in a unit test for 2.5 hours. Google wanst much help either.
But then again, I'm relatively unexperienced bootstrapping and mocking core laravel components.

I could take a jab at it if I could test those queuedClosures working and triggereng the closure on event.

Reply:
I suppose that's fine for now then, I don't really know anything about closure based event listeners in the first place. Feel free to toss a TODO comment in there with references to further information and we can take another look at it closer to 1.2

Reply:
The normal closures are pretty straight forward. They just run when the condition is triggered when the event is fired with the appropriate type hint(think function(HoppingBunnyEvent $event){});,   
Not that different from the string based event handlers, except you can completely bypass the appropriate string finding and use your ide's autocomplete to find the correct class. 

It's the queued closure that's tricky. It get scheduled as a job, which then presumably gets run when `php artisan queue:work --queue=high,default` is called.

So all the parts that are involved with scheduling tasks/jobs need to be booted for a proper test so it can ben known it works and is accepted properly and registered properly etc...  Especialy when we want to add priorities that sort the jobs.

Then there are two queues high, low it seems, https://laravel.com/docs/8.x/queues#queue-priorities, need to research if those can define even more queues so we can implement a sort of priority queue where they can be sorted into, or simply just only accept high/low as parameters when it's a queued job.

Reply:
I don't think we'd want to support priorities on queued event listeners, it sounds like it's more so for the use case of "hey, use this event being fired as a trigger that I should take care of a job later" rather than actually looking to do anything to filter or modify existing behavior in an event handler.

Reply:
Okay, that makes it easy then. Queueables get pushed to parent, closure event listeners get priority treatment. I'll see what I can do :-)

Reply:
So uhm, I did it, I looked at the laravel code with fresh eyes and saw how it could be easily implemented into our event handler queue without breaking stuff.

However, I did add a [parameter description](https://github.com/wintercms/storm/pull/26/commits/1cda0ebc75eec61ce1c0fa37c73f277370697a55#diff-1c29f9614d646d36e4f57a6a43cdcbf485f88eeae3a6fa929355f4224d536299R31) describing the duality of one parameter when a Closure type is passed.

Did I describe it correctly, it is this undesired behaviour and should they pass a null  and then priority?

so do you want:
```php
$dispatcher->listen(function(Event $foo){}, null, 2);
```
or do you want(what I've currently coded in)

```php
$dispatcher->listen(function(Event $foo){}, 2);
```

Reply:
A queuedevent  when initialized gets the closure that's passed to it assigend to the property closure.
Then for the event handler the method QueuedClosure::resolve() is called.
Resolve wraps it in a new closure which is returned. 
This resolved closure injects it into the queue scheduler when it is executed https://github.com/laravel/framework/blob/29b019774b746c90534465f1b949767707d49245/src/Illuminate/Events/QueuedClosure.php#L115
When the event class is triggered it is this resolved closure/job injector that is executed, to schedule the passed event closure.